theano/gpuarray/multinomial.py

```diff
@@ -247,3 +487,19 @@ def local_gpua_multinomial(node, context_name):
         gpu_op = GPUAMultinomialFromUniform(node.op.odtype)
         return gpuarray.elemwise.GpuDimShuffle([False, False], [1, 0])(
             gpu_op(p, u))
+
+@register_opt()
+@op_lifter([theano.sandbox.multinomial.MultinomialWOReplacementFromUniform])
+def local_gpua_multinomial_wor(node, context_name):
+    # TODO : need description for function
+    p, u, n = node.inputs
+    # try:
+    #     if get_scalar_constant_value(n_samples) != 1:
+    #         return None
+    # except NotScalarConstantError:
+    #     return None
+    m, = node.outputs
+    if ((p.dtype == u.dtype == 'float32') and (m.dtype == 'int64')):
+        gpu_op = GPUAMultinomialWOReplacementFromUniform(node.op.odtype)
+        return gpuarray.elemwise.GpuDimShuffle([False, False], [1, 0])(
```

Comment:
This is something I picked up from the with-replacement code. My understanding is that it makes memory fetching more efficient.